src/utilities/structured-output/structured-output.factory.ts

```diff
@@ -0,0 +1,169 @@
+import { OASResult, OperationExampleResult } from '../../results';
+import { StructuredOutput } from './structured-output.interface';
+
+class StructuredOutputFactory {
+  static buildFromOASResult(input: OASResult): StructuredOutput {
+    const testResults = input.results;
+
+    // assemble authenticationType string
+    const authenticationType = input.securitySchemes
+      .map((x) => x.type)
+      .join('/');
+
+    // assemble top-level output
+    const output: StructuredOutput = {
+      id: input.testName,
+      config: {
+        oasPath: String(input.oasPath),
+        server: String(input.server),
+        authenticationType: String(authenticationType),
+      },
+      error: input.error,
+      results: undefined,
+    };
+
+    if (!input.error && testResults) {
+      // assemble api-level output
+      this.assembleApiLevel(testResults, output);
+    }
+
+    return output;
+  }
+
+  private static assembleApiLevel(
+    testResults: OperationExampleResult[],
+    output: StructuredOutput,
+  ): void {
+    // calculate pass/warn/fail totals
+    const passCount = testResults.filter(
+      (result) => result.failures.size === 0,
```

Comment:
Pass means no failures or warnings

Reply:
Same for line 82

Reply:
This is not how LOAST has been operating, so far pass has meant no failures (warnings are fine).  It's easy enough to change, but will impact both regular and json outputs, and probably needs its own ticket.